Improve trace status handling (#3214)
open-telemetry/opentelemetry-go: CHANGELOG.md

- The metric portion of the OpenCensus bridge (`go.opentelemetry.io/otel/bridge/opencensus`) has been reintroduced. (#3192)
- The OpenCensus bridge example (`go.opentelemetry.io/otel/example/opencensus`) has been reintroduced. (#3206)

### Changed

> - `span.SetStatus` has been updated to comply with the OpenTelemetry specification.
>   Calls that lower the status are now noops. (#3214)

Reviewer: ```suggestion
- `span.SetStatus` has been updated such that calls that lower the status are now no-ops. (#3214)
```

The existing behavior was compliant, since the spec said attempts to lower the status value should be ignored, not that they must be ignored.
